### PR TITLE
Wallet: add loadFromFile method that supports WalletTool

### PR DESCRIPTION
**Pull request by msgilligan**
* Add `Wallet.loadFromFile` that takes options needed by WalletTool
* Add `Wallet.loadFromFileStream` that supports the extra options
* Update `WalletTool` to use the new `loadFromFile` method

### CODE REVIEW

File: core/src/main/java/org/bitcoinj/wallet/Wallet.java [hunk 1]
```diff
@@ -1668,12 +1668,25 @@ public Context getContext() {
...
      */
     public static Wallet loadFromFile(File file, @Nullable WalletExtension... walletExtensions) throws UnreadableWalletException {
-        try {
-            try (FileInputStream stream = new FileInputStream(file)) {
-                return loadFromFileStream(stream, walletExtensions);
-            }
+        return loadFromFile(file, false, false, walletExtensions);
+    }
+
+    /**
```
Comment (schildbach) [inline core/src/main/java/org/bitcoinj/wallet/Wallet.java:1678]: There is an incomplete JavaDoc here.

**Review by schildbach — COMMENTED** [review]
Generally looks good. Except:

* The JavaDocs need work
* It would be great to have unit tests for the new API

Also I wonder if we should perhaps drop wallet extensions?

**Comment by msgilligan** [comment]
> * The JavaDocs need work
> * It would be great to have unit tests for the new API


Yes, that's why it's a DRAFT 😁. I will add them.

> Also I wonder if we should perhaps drop wallet extensions?

I think there is one use of wallet extensions in the JavaFX wallet. We'll need to remove that first. Are there any other existing uses that you know of?

File: core/src/main/java/org/bitcoinj/wallet/Wallet.java (hunk 1 above)
Comment (msgilligan) [reply core/src/main/java/org/bitcoinj/wallet/Wallet.java:1678]: Fixed.

**Comment by msgilligan** [comment]
@schildbach Actually, I think it would be better to add the tests in a separate PR. I'm hoping we can develop these tests with JUnit 5, so we need to pick up that discussion and see if we can figure out a way to use JUnit 5 for at least some subset of tests.  See https://github.com/bitcoinj/bitcoinj/pull/2022 -- but I think what we should do is add a integration-tests subproject to do this in. We could then check in a few `Wallet` data files and use them for testing these new methods _and_ for testing `WalletTool`.

**Comment by schildbach** [comment]
> I think there is one use of wallet extensions in the JavaFX wallet. We'll need to remove that first. Are there any other existing uses that you know of?

I'm not aware of any.

File: core/src/main/java/org/bitcoinj/wallet/Wallet.java (hunk 1 above)
Comment (schildbach) [reply core/src/main/java/org/bitcoinj/wallet/Wallet.java:1678]: It's still missing the overall description and the description of the `ignoreMandatoryExtensions` parameter.

Comment (msgilligan) [reply core/src/main/java/org/bitcoinj/wallet/Wallet.java:1678]: That should be resolved. For some reason that latest force push is not showing here. I'm going to mark resolved.

**Review by msgilligan — COMMENTED** [review]
OK, the remaining JavaDoc issues have been resolved and force-pushed. Should be ready re-review/merge.

File: core/src/main/java/org/bitcoinj/wallet/Wallet.java (hunk 1 above)
Comment (msgilligan) [reply core/src/main/java/org/bitcoinj/wallet/Wallet.java:1678]: Ignore my previous confusion, I'll fix and force push...

**Comment by schildbach** [comment]
Merged, except the one small change to the `call()` exception declaration. It doesn't feel related to this PR to me.

**Comment by msgilligan** [comment]
> except the one small change to the `call()` exception declaration. It doesn't feel related to this PR to me.

OK, here's that change in a separate PR: https://github.com/bitcoinj/bitcoinj/pull/2387

**Pull request closed without merging**